ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -111,8 +133,11 @@ public void comprehensiveTest() throws IOException, StoreException {
       }
       long lastEntrySize = Utils.getRandomLong(TestUtils.RANDOM, 1000) + 1;
       long endOffset = offsets.get(offsets.size() - 1) + lastEntrySize;
-      addPutEntries(offsets, lastEntrySize, indexSegment, referenceIndex);
-      verifyIndexSegmentDetails(indexSegment, startOffset, offsets.size(), false, endOffset);
+      ArrayList<IndexEntry> newEntries = addPutEntries(offsets, lastEntrySize, indexSegment, referenceIndex);
+      resetKey = new Pair<>(newEntries.get(0).getKey(),
+          (newEntries.get(0).getValue().isFlagSet(IndexValue.Flags.Delete_Index))
+              ? PersistentIndex.IndexEntryType.DELETE : PersistentIndex.IndexEntryType.PUT);
```

Comment:
since you are adding only puts, is there any scenario in which this can be a delete entry?